src/main/java/io/debezium/connector/vitess/VitessConnector.java

```diff
@@ -54,4 +60,29 @@ public ConfigDef config() {
     public String version() {
         return Module.version();
     }
+
+    @Override
+    protected void validateConnection(Map<String, ConfigValue> configValues, Configuration config) {
+        ConfigValue hostnameValue = configValues.get(RelationalDatabaseConnectorConfig.HOSTNAME.name());
+        // Try to connect to the database ...
+        final VitessConnectorConfig connectionConfig = new VitessConnectorConfig(config);
+        try (VitessReplicationConnection connection = new VitessReplicationConnection(connectionConfig, null)) {
+            try {
+                connection.execute("SHOW DATABASES");
+                LOGGER.info("Successfully tested connection for {} with user '{}'", connection.connectionString(), connection.username());
+            }
+            catch (StatusRuntimeException e) {
+                LOGGER.info("Failed testing connection for {} with user '{}'", connection.connectionString(), connection.username());
+                hostnameValue.addErrorMessage("Unable to connect: " + e.getMessage());
+            }
+        }
+        catch (Exception e) {
+            LOGGER.error("Unexpected error shutting down the database connection", e);
```

Comment:
```suggestion
            LOGGER.error("Unexpected error validating the database connection", e);
```

I think that's what it should say? And an error message should be returned to the validation consumer?

Reply:
Good catch! Addressed